### ISSUE TITLE
Handle pending jobs for renamed subscription jobs

### ISSUE THREAD

**Issue by kristinalim** [openfoodfoundation/openfoodnetwork#3643]
## Description

In [this commit](https://github.com/openfoodfoundation/openfoodnetwork/commit/a9b5fd69d883bf6dfd49628b6fec4993185ef4b3), seeing that the following jobs for subscriptions have been renamed:

- `SubscriptionConfirmJob`
- `SubscriptionPlacementJob`

Based on this [review comment](https://github.com/openfoodfoundation/openfoodnetwork/pull/3609#discussion_r267183802), we are not deleting all pending jobs. Which means that pending jobs of above type will not have a corresponding class anymore in v2.

We need to address this.

Options I see are:

* Add back the old job classes, but deprecate them and just have them use the logic in the new job classes. Then remove them in a later version of OFN.
* When migrating, have longer downtime and wait for all pending jobs to complete.

## Expected Behavior

A job of the above types would still be executed after migration to v2.

## Actual Behaviour

It would not be executed.

## Context

This came up in #3533.

**Comment by luisramos0** [comment]
@kristinalim those jobs were renamed in v1 and the new names are already in production. I dont understand how this is a Spree Upgrade issue :-)

**Comment by kristinalim** [comment]
@luisramos0 Okay, I got the rename direction wrong and had a major logic lapse.

That commit is too old, I feel it will be more dangerous to reenable whatever background jobs that stopped working. But I will check a couple of production servers before closing this issue.

**Comment by kristinalim** [comment]
I ran the following in all the managed instances (AU, DE, BE, FR, ES, UK, CA, US) and there are no matches:

```ruby
Delayed::Job.count # Just check that the next line will not use up too much resources.
Delayed::Job.all.select { |job| ["StandingOrderConfirmJob", "StandingOrderPlacementJob"].include? job.name }
```

Closing this, as this isn't actually be an issue.

For future job renames, we should make sure to let all pending jobs of the old type finish first or provide a transitional job class. But probably this wasn't a concern before because subscriptions were/are not widely used.
